### DISCUSSION TITLE
set default value for Ternary filter: is used, but not visible in filter selector?

### DISCUSSION THREAD

**Discussion by wivaku** [filamentphp/filament#5127] [Help] [answered]
I have [set a default value](https://filamentphp.com/docs/2.x/tables/filters#setting-default-values
) for a [Ternary filter](https://filamentphp.com/docs/2.x/tables/filters#ternary-filters):
```
            ->filters([
                Tables\Filters\TernaryFilter::make('is_future')
                    ->queries(
                        true: fn($query) => $query->future(),
                        false: fn($query) => $query->notFuture(),
                    )
                    ->default(true)
            ], layout: Layout::AboveContent)
```

When I reload the page I see that the filter is active (`Active filters "Is future: Yes X"`), but the filter selector shows `-`.
Is this a bug?

Using v2.16.52.

**Comment by danharrin** [comment, marked as answer]
Hmm, try `default(1)`. We might need to somehow translate this just for ternary filters

**Reply by wivaku** [reply]
That was it indeed, thanks! Interesting. The query string in both cases is showing `...?tableFilters[is_future][value]=1` (so: that one correctly interprets `true`) and the actual filter is applied correctly when using `true`, it is only the form field that behaves differently.
